### PR TITLE
[Improve][e2e]change the jdbc driver scope to provider

### PR DESCRIPTION
**Pull request by liugddx**
## Purpose of this pull request

change the jdbc driver scope to provider

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by liugddx** [comment]
subtask of #2642 .change the jdbc driver scope to provider and improve the jdbc connector doc.
PTAL @ashulin @Hisoka-X @TyrantLucifer thanks

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcDmdbIT.java [hunk 1]
```diff
@@ -144,4 +145,9 @@ public void testJdbcDmdbSourceAndSink() throws IOException, InterruptedException
         assertHasData(SINK_TABLE);
     }
 
+    @Override
+    protected void executeExtraCommands(GenericContainer<?> container) throws IOException, InterruptedException {
+        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/flink/seatunnel/plugins/Jdbc/lib && cd /tmp/flink/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcDmdbIT.java]: Use dir `/tmp/seatunnel/plugins/`?

reference
#2762

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcPhoenixIT.java [hunk 2]
```diff
@@ -177,4 +178,10 @@ private void batchInsertData() throws SQLException, ClassNotFoundException {
             throw e;
         }
     }
+
+    @Override
+    protected void executeExtraCommands(GenericContainer<?> container) throws IOException, InterruptedException {
+        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/flink/seatunnel/plugins/Jdbc/lib && cd /tmp/flink/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcPhoenixIT.java]: as above

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-jdbc-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/jdbc/JdbcDmdbIT.java [hunk 3]
```diff
@@ -148,4 +148,10 @@ public void testDMDBSourceToJdbcSink() throws SQLException, IOException, Interru
         assertHasData(SINK_TABLE);
     }
 
+    @Override
+    protected void executeExtraCommands(GenericContainer<?> container) throws IOException, InterruptedException {
+        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/spark/seatunnel/plugins/Jdbc/lib && cd /tmp/spark/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-jdbc-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/jdbc/JdbcDmdbIT.java]: as above

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-jdbc-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/jdbc/JdbcPhoenixIT.java [hunk 4]
```diff
@@ -177,4 +178,10 @@ private void batchInsertData() throws SQLException, ClassNotFoundException {
             throw e;
         }
     }
+
+    @Override
+    protected void executeExtraCommands(GenericContainer<?> container) throws IOException, InterruptedException {
+        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/spark/seatunnel/plugins/Jdbc/lib && cd /tmp/spark/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-jdbc-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/jdbc/JdbcPhoenixIT.java]: as above

**Comment by liugddx** [comment]
PTAL @hailin0

File: docs/en/connector-v2/sink/Jdbc.md [hunk 5]
```diff
@@ -84,6 +84,7 @@ there are some reference value for params above.
 | mysql      | com.mysql.cj.jdbc.Driver | jdbc:mysql://localhost:3306/test          | com.mysql.cj.jdbc.MysqlXADataSource | https://mvnrepository.com/artifact/mysql/mysql-connector-java |
 | postgresql | org.postgresql.Driver    | jdbc:postgresql://localhost:5432/postgres | org.postgresql.xa.PGXADataSource    | https://mvnrepository.com/artifact/org.postgresql/postgresql  |                                                             |
 | dm         | dm.jdbc.driver.DmDriver  | jdbc:dm://localhost:5236                  | dm.jdbc.driver.DmdbXADataSource     | https://mvnrepository.com/artifact/com.dameng/DmJdbcDriver18  |
+| phoenix         | org.apache.phoenix.queryserver.client.Driver  | jdbc:phoenix:thin:url=http://localhost:8765;serialization=PROTOBUF                  | /     | https://mvnrepository.com/artifact/com.aliyun.phoenix/ali-phoenix-shaded-thin-client  |
```
Comment (ashulin) [inline docs/en/connector-v2/sink/Jdbc.md]: check markdown style

File: seatunnel-connectors-v2/connector-jdbc/pom.xml [hunk 6]
```diff
@@ -29,13 +29,6 @@
 
     <artifactId>connector-jdbc</artifactId>
 
-    <properties>
-        <phoenix.version>5.2.5-HBase-2.x</phoenix.version>
-        <mysql.version>8.0.16</mysql.version>
-        <postgresql.version>42.3.3</postgresql.version>
-        <dm-jdbc.version>8.1.2.141</dm-jdbc.version>
-    </properties>
-
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-jdbc/pom.xml]: Don‘t manage the properties of the connector in root pom.xml

That's my suggestion:

```xml
    <properties>
        <phoenix.version>5.2.5-HBase-2.x</phoenix.version>
        <mysql.version>8.0.16</mysql.version>
        <postgresql.version>42.3.3</postgresql.version>
        <dm-jdbc.version>8.1.2.141</dm-jdbc.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>com.aliyun.phoenix</groupId>
            <artifactId>ali-phoenix-shaded-thin-client</artifactId>
        </dependency>
    </dependencies>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.aliyun.phoenix</groupId>
                <artifactId>ali-phoenix-shaded-thin-client</artifactId>
                <version>${phoenix.version}</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

This is then used in the E2E module

```xml
    <dependencies>
        <dependency>
            <groupId>com.aliyun.phoenix</groupId>
            <artifactId>ali-phoenix-shaded-thin-client</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.apache.seatunnel</groupId>
                <artifactId>connector-jdbc</artifactId>
                <version>${project.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcDmdbIT.java [hunk 7]
```diff
@@ -144,4 +145,9 @@ public void testJdbcDmdbSourceAndSink() throws IOException, InterruptedException
         assertHasData(SINK_TABLE);
     }
 
+    @Override
+    protected void executeExtraCommands(GenericContainer<?> container) throws IOException, InterruptedException {
+        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/seatunnel/plugins/Jdbc/lib && cd /tmp/flink/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
Comment (ashulin) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcDmdbIT.java]: Other similar code in this PR is handled like this: 
```suggestion
        Container.ExecResult extraCommands = container.execInContainer("bash", "-c", "mkdir -p /tmp/seatunnel/plugins/Jdbc/lib && cd /tmp/seatunnel/plugins/Jdbc/lib && curl -O " + THIRD_PARTY_PLUGINS_URL);
```
To support multiple engines, I unified the Seatunnel Home(`/tmp/seatunnel`) without engine prefixes

**Comment by liugddx** [comment]
@ashulin  i fixed those problems, and PTAL again ,thanks.

File: seatunnel-connectors-v2/connector-jdbc/pom.xml [hunk 8]
```diff
@@ -30,12 +30,40 @@
...
+                <groupId>org.postgresql</groupId>
+                <artifactId>postgresql</artifactId>
+                <version>${postgresql.version}</version>
+                <scope>provided</scope>
+            </dependency>
+            <dependency>
+                <groupId>com.dameng</groupId>
+                <artifactId>DmJdbcDriver18</artifactId>
+                <version>${dm-jdbc.version}</version>
+            </dependency>
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-jdbc/pom.xml]:
```suggestion
            <dependency>
                <groupId>com.dameng</groupId>
                <artifactId>DmJdbcDriver18</artifactId>
                <version>${dm-jdbc.version}</version>
                <scope>provided</scope>
            </dependency>
```
Then delete version from <version># line83

**Comment by liugddx** [comment]
@ashulin @TyrantLucifer @CalvinKirs @Hisoka-X @EricJoy2048 PTAL thanks.

**Review by ashulin — DISMISSED** [review]
LGTM

**Review by ashulin — DISMISSED** [review]

**Comment by liugddx** [comment]
@TyrantLucifer  @EricJoy2048  PTAL thanks

**Comment by ic4y** [comment]
Need to add driver package download for all JDBC inheritance tests
E.g XaGroupOpsImplIT.testRecoverAndRollback

**Comment by liugddx** [comment]
Done. @ic4y  .please rerun CI 
image
@EricJoy2048 thanks.

File: seatunnel-e2e/pom.xml [hunk 9]
```diff
@@ -40,6 +40,17 @@
         <junit4.version>4.13.2</junit4.version>
         <maven-jar-plugin.version>2.4</maven-jar-plugin.version>
     </properties>
+    <dependencyManagement>
```
Comment (Hisoka-X) [inline seatunnel-e2e/pom.xml]: Why e2e root pom need add `connector-jdbc`'s dependency?

Comment (liugddx) [reply seatunnel-e2e/pom.xml]: The driver jar is used by connector and e2e,so i want a unified version

Comment (Hisoka-X) [reply seatunnel-e2e/pom.xml]: Just import it in `seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e` and `seatunnel-spark-connector-v2-e2e/connector-jdbc-spark-e2e`, not here

Comment (liugddx) [reply seatunnel-e2e/pom.xml]: Done.

Comment (Hisoka-X) [reply seatunnel-e2e/pom.xml]: You need fixed ci

Comment (liugddx) [reply seatunnel-e2e/pom.xml]: > 

mysql and postgres e2e fail.and i have fixed it.

**Review by EricJoy2048 — APPROVED** [review]
LGTM @hailin0  @TyrantLucifer  @ic4y  PTAL

**Comment by EricJoy2048** [comment]
> LGTM

**Comment by liugddx** [comment]
please rerun CI. @EricJoy2048

**Comment by liugddx** [comment]
@TyrantLucifer  PTAL

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**